comments_async/serializer & tests/comments: serializer treats blocked comments s…

sabinammm opened the pull request:
…ame as deleted comments

fixes #899
also fixes [#166 in kosmo](https://github.com/liqd/a4-kosmo/issues/166)

fuzzylogic2000 reviewed (commented):
Yes! That's nearly it.
The only thing left is the categories. In a debate, that is still shown for the blocked comment, but not the other two. Let me know if you have a hard time finding that!
![Screenshot from 2022-02-01 11-35-08](https://user-images.githubusercontent.com/8178179/151953351-2b856d29-a2a6-4171-bc6a-333f099e354a.png)

fuzzylogic2000 commented on adhocracy4/comments_async/serializers.py:
```diff
@@ -40,7 +40,10 @@ def to_representation(self, instance):
         categories = {}
-        if ret['comment_categories']:
+        if ret['is_blocked']:
+            ret['comment_categories'] = categories
```
I don't really get, why you put it here and moved the if and made the other one to an `elif`.

fuzzylogic2000 reviewed (changes requested):
Cool, that you found where it happens. I would solve it a bit differently, though. Let's have a look together tomorrow.

fuzzylogic2000 commented:
@sabinammm I removed the bit with the categories again and added another commit that changes it to returning the empty dict when the comment is blocked.

sabinammm commented:
> @sabinammm I removed the bit with the categories again and added another commit that changes it to returning the empty dict when the comment is blocked.

Ok, I see, thank you!!
